src/mapeo-project.js

```diff
@@ -548,6 +549,100 @@ export class MapeoProject extends TypedEmitter {
   get $icons() {
     return this.#iconApi
   }
+
+  /** @param {Object} opts
+   *  @param {string} opts.configPath
+   */
+  async importConfig({ configPath }) {
+    // check for already present fields and presets and delete them if exist
+    const presets = await this.preset.getMany()
+    if (presets.length !== 0) {
+      for (const preset of presets) {
+        const { fieldIds, versionId: presetVersionId } = preset
+        // delete fields
+        for (const fieldId of fieldIds) {
+          const field = await this.field.getByDocId(fieldId)
+          const { deleted } = await this.field.delete(field.versionId)
+          if (!deleted) throw new Error('error deleting field from db')
+        }
+        // delete preset
+        const { deleted: deletedPreset } = await this.preset.delete(
+          presetVersionId
+        )
+        if (!deletedPreset) throw new Error('error deleting preset from db')
```

Comment:
If there is an error deleting the preset, then `preset.delete()` will throw. `deletedPreset` being false here would indicate that there is a bug in the code, so this is code to write in a test, not runtime code.